ddl/ddl_worker_test.go

```diff
@@ -469,16 +469,25 @@ func (s *testDDLSuite) TestCancelJob(c *C) {
 	firstJobID := job.ID
 	tests := buildCancelJobTests(firstJobID)
 	var checkErr error
+	var mu sync.RWMutex
 	var test *testCancelJob
+	updateTest := func(t *testCancelJob) {
+		mu.Lock()
+		test = t
+		mu.Unlock()
+	}
 	hookCancelFunc := func(job *model.Job) {
 		if job.State == model.JobStateSynced || job.State == model.JobStateCancelled || job.State == model.JobStateCancelling {
 			return
 		}
 		// This hook only valid for the related test job.
 		// This is use to avoid parallel test fail.
+		mu.RLock()
```

Comment:
Why use Read-write lock , rather than mutex lock ?

Reply:
em... Actually in this scene, use mutex and read-write mutex is same. because there is only 2 goroutines, 1 goroutine write the variable, other goroutine read the variable.

Reply:
then， use mutex is better.

Reply:
ok. done.